testutils/check-calling-form-in-baseline.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const usage = `Reads calling forms from gapic_yaml_file and make sure each is used by
+some *.baseline file in dir.
+The line specifying the calling form is in format
+
+  callingFormCheck: <id> <language>: <callingForm1> <callingForm2> ...
+
+The string "callingFormCheck" must start the line, preceded only by a possilby empty
+run of whitespaces.
+`
+
+func main() {
+	yamlFname := flag.String("yaml", "", "gapic yaml file")
+	flag.Usage = func() {
+		binName := os.Args[0]
+		out := os.Stderr // Should be flag.CommandLine.Output(), but Travis is too old
+		fmt.Fprintf(out, "Usage of %s:\n", binName)
+		fmt.Fprintf(out, "%s -yaml <gapic_yaml_file> [dir]\n", binName)
+		flag.PrintDefaults()
+		fmt.Fprint(out, usage)
+	}
+	flag.Parse()
+
+	if *yamlFname == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+	yamlf, err := os.Open(*yamlFname)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer yamlf.Close()
+
+	checks, err := readChecks(yamlf)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if len(checks) == 0 {
+			return filepath.SkipDir
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		if filepath.Ext(path) != ".baseline" {
+			return nil
+		}
+		return delFoundForms(path, checks)
+	}
+	dir := "."
+	if flag.NArg() > 0 {
+		dir = flag.Arg(0)
+	}
+	if err := filepath.Walk(dir, walkFn); err != nil {
+		log.Fatal(err)
+	}
+
+	if len(checks) > 0 {
+		var errs []checkConfig
+		for c := range checks {
+			errs = append(errs, c)
+		}
+		sort.Sort(checkConfigSlice(errs)) // Not sort.Slice, Travis is too old.
+		fmt.Println("not found:", errs)
+		os.Exit(1)
+	}
+}
+
+type checkConfigSlice []checkConfig
```

Comment:
I'd lean towards moving everything from here down to a separate file, especially in light of adding tests.

Reply:
I personally have a bias towards keeping `main()` and main.go on the small side, mostly piping arguments from the command line into libraries that do the real work. But you can leave as is if you prefer.